webui/react/src/pages/ExperimentDetails/CodeViewer/CodeViewer.tsx

```diff
@@ -0,0 +1,269 @@
+  });
+  const editorClasses = classNames({
+    [ css.hideElement ]:
+    (documentWidth <= 1024) && (viewMode === 'tree'),
+  });
+
+  // map the file tree
+  useEffect(() => {
+    try {
+      (async () => {
+        const files = await getExperimentFileTree({ experimentId });
+
+        const navigateTree = (node: FileNode, key: string): DataNode => {
+          treeMap.set(key, node.path || '');
+
+          const newNode: DataNode = {
+            className: 'treeNode',
+            isLeaf: true,
+            key,
+            title: node.name,
+          };
+
+          if (node.files?.length) {
+            newNode.children = node.files.map((chNode, idx) => navigateTree(chNode, `${key}-${idx}`));
+            newNode.isLeaf = false;
+          }
+
+          return newNode;
+        };
+
+        if (config) {
+          setFileTree([
+            {
+              className: 'treeNode',
+              isLeaf: true,
+              key: '0-0',
+              title: 'Configuration',
+            },
+            ...files.map<DataNode>((node, idx) => navigateTree(node, `0-${idx + 1}`)),
+          ]);
+
+          setFileInfo({
+            data: yaml.dump(config),
+            name: 'Configuration',
+            path: 'Configuration',
+          });
+
+          if (documentWidth <= 1024) { // if it's in mobile view and we have a config file available, render it as default
+            setViewMode('editor');
+          }
+        } else {
+          setFileTree(files.map<DataNode>((node, idx) => navigateTree(node, `0-${idx}`)));
+        }
+      })();
+    } catch (error) {
+      throw new Error(error as string);
+    }
+  }, [ treeMap, config, documentWidth, experimentId ]);
+
+  const onSelectFile = async (
+    keys: React.Key[],
+    info: { [key:string]: unknown, node: DataNode },
+  ) => {
+    if (info.node.title === 'Configuration') {
+      setFileInfo({
+        data: yaml.dump(config),
+        name: 'Configuration',
+        path: 'Configuration',
+      });
+
+      return;
+    }
+
+    const filePath = treeMap.get(String(keys[0])) as string;
+
+    if (filePath.includes('.')) { // check if the selected node is a file
+      setIsFetching(true);
+
+      try {
+        const file = await getExperimentFileFromTree({ experimentId, filePath });
+
+        setIsFetching(false);
+        setFileInfo({
+          data: decodeURIComponent(escape(window.atob(file))),
+          name: info.node.title as string,
+          path: filePath,
+        });
+
+        if (documentWidth <= 1024) {
+          setViewMode('editor');
+        }
+      } catch (error) {
+        setIsFetching(false);
+
+        throw new Error(error as string);
+      }
+    }
+  };
+
+  const setEditorLanguageSyntax = () => {
+    const fileExt = (fileInfo?.path || '').split('.')[1];
+
+    if (fileExt === 'md') {
+      return 'markdown';
+    }
+
+    if (fileExt === 'ts') {
+      return 'typescript';
+    }
+
```

Comment:
can we get highlighting for `.py` files? those would be the most useful.

Reply:
i think probably better to remove and remove from the `craco.config.js` since presumably it increases bundle size unnecessarily?

also might be better to just do yaml for `.yaml` files and plaintext or whatever as the default, unless you can check that it handles okay when it's not a `.yaml`